### ISSUE TITLE
Warning about dbus-policy

### ISSUE THREAD

**Issue by agaida** [andrew-bibb/cmst#111]
Hi, 
running lintian over the package this warning came up:
W: cmst: dbus-policy-without-send-destination etc/dbus-1/system.d/org.cmst.roothelper.conf <policy group="network"><allow send_interface="org.cmst.roothelper"/>

Afaik this should be avoided.
Rationale: https://lintian.debian.org/tags/dbus-policy-without-send-destination.html

Greetz Alf

**Comment by andrew-bibb** [comment]
A couple of things, one is that in getting some translations started I discovered a bunch of problems with some of the browser select and agent dialog code.  I actually have no clue how the program was even building, but I I think I've got most of it cleaned up (and uploaded) now.  Don't want you to package something that does not work.

To the question at hand.  The file you getting the error for is actually distro specific.  If a distro is not specified at qmake with the default being Arch linux (see the building and installing txt file).  Is the file working for you? Right now there are files for Arch and Slackware, easy enough for me to put in another for Debian, even if it is just a copy of the Arch linux one.  The biggest difference between the two is the name of the "network" or "netdev" group.  

Lastly, do you know what line needs to be changed:  The default file follows:
```
<?xml version="1.0" encoding="UTF-8"?> <!-- -*- XML -*- -->

<!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
 "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
<busconfig>
    <!-- Only root can own roothelper -->
    <policy user="root">
        <allow own="org.cmst.roothelper"/>
        <allow send_destination="org.cmst.roothelper"/>
        <allow send_interface="org.cmst.roothelper"/>
    </policy>
    <policy group="network">
        <allow send_destination="org.cmst.roothelper"/>
        <allow send_interface="org.cmst.roothelper"/>
    </policy>
</busconfig>
```

As you can see I have send_destination listed and according to the link you had you get the message when you don't.  I'm really not quite sure what I need to do here.

**Comment by agaida** [comment]
1) if things are broken on checkout - no problem, as long it isn't the release :)
2) ok, i will try this out, dbus was not a field of special interest right now for me

**Comment by bubbless** [comment]
I got a Dbus error when I connect a Wireless AP.
```
Error Name: org.freedesktop.DBus.Error.NoReply

Error Message: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```

**Comment by andrew-bibb** [comment]
bubbless,

Your issue is definetely something different.  This thread is about issues during build, yours is during run.  Feel free to open a new issue about yours and link back to here if you want.  We've got plenty of numbers for new issues.

If you do I'd like to know the CMST version number you are running (About button in the Help tab).  I did have a period of time not too long ago when I was checking too quickly for a reply when calling connect().  That should be fixed in the current version and also some of the more recent GIT versions.

I'd also like to see if you can connect manually.  From a terminal please type:

    connmanctl
    services
    connect <your wifi service>
    quit

Give it a couple of seconds connect.  In the connect command tab completion works, so typing something like:
`connect wifi<tab>` should should complete the command. 

I want to see if it works from the internal Connman command line interface.  That will tell us if the problem is at my end or the Connman end.

**Comment by bubbless** [comment]
@andrew-bibb  I running connmanctl manually, and there are no error.
````
connmanctl> connect wifi_xxxxxxxxxx_managed_psk
Connected wifiwifi_xxxxxxxxxx_managed_psk
```
No alert dbus error window.
How can I echo the debugging information?

**Comment by andrew-bibb** [comment]
Really sounds like something I did and not Connman.  I'm going to ask if you could open a new issue here and reference this thread back.  This is definetely a different issue so I'd like to keep the separate.

When you do post can you get me the version and release date of the program you are using.  Ou can find this in the Help tab, use the CMST button in the About area.  As I mentioned above, I did a fix in this area not too long ago and I want to verify which side of the fix you are dealing with.

**Comment by andrew-bibb** [comment]
I think this was resolved in the pull request for issue #154
